spec/integrations/base_spec.rb

```diff
@@ -85,6 +85,19 @@
           end
           it('returns json with association') { should eq(result) }
         end
+        context "Given association with dynamic blueprint" do
+          let(:blueprint) do
+            Class.new(Blueprinter::Base) do
+              association :user, blueprint: ->(obj) { user_blueprint }
+              def self.user_blueprint
+                Class.new(Blueprinter::Base) { identifier :id }
+              end
+            end
+          end
```

Comment:
Would you redo the test so that it follows more closely what the use case should look like? For example: 
`association :user, blueprint: ->(obj) { obj.blueprint }`

Reply:
I was having a very hard time making the user/vehicle object respond to a blueprint method. Since the feature was to let association blueprint be fetched dynamically, I took this route instead.

Reply:
You should be able to reopen the User class and add `blueprint` to it.

Try this:
```ruby
        context "Given association with dynamic blueprint" do
          class UserBlueprint < Blueprinter::Base
            fields :id
          end
          class User < ActiveRecord::Base
            def blueprint
              UserBlueprint
            end
          end
          let(:blueprint) do
            Class.new(Blueprinter::Base) do
              association :user, blueprint: ->(obj) { obj.blueprint }
            end
          end
          it "should render the association with dynamic blueprint" do
            expect(JSON.parse(blueprint.render(vehicle))["user"]).to eq({"id"=>obj.id})
          end
        end
```

Reply:
I tried various ways, `patching`, `class_eval` etc. Nothing helped, so gave up. I'll give this another shot today! 😄